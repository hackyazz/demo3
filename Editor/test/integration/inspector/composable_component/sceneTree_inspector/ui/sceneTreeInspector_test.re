open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

let _ =
  describe("SceneTreeInspector", () => {
    let sandbox = getSandboxDefaultVal();

    beforeEach(() => {
      sandbox := createSandbox();
      MainEditorSceneTool.initState(~sandbox, ());

      EventListenerTool.buildFakeDom()
      |> EventListenerTool.stubGetElementByIdReturnFakeDom;
    });
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));

    describe("prepare currentSelectSource is SceneTree", () => {
      beforeEach(() =>
        CurrentSelectSourceEditorService.setCurrentSelectSource(
          SceneTreeWidgetService.getWidget(),
        )
        |> StateLogicService.getAndSetEditorState
      );

      describe("test rename gameObject", () => {
        beforeEach(() =>
          MainEditorSceneTool.createDefaultScene(
            sandbox,
            MainEditorSceneTool.setFirstCubeToBeCurrentSceneTreeNode,
          )
        );

        describe("test logic", () =>
          describe("test engine", () => {
            let _getName = () =>
              GameObjectEngineService.unsafeGetGameObjectName(
                SceneTreeEditorService.unsafeGetCurrentSceneTreeNode
                |> StateLogicService.getEditorState,
              )
              |> StateLogicService.getEngineStateToGetData;

            test("test rename gameObject ", () => {
              let newName = "gameObjectNewName";

              SceneTreeInspectorTool.renameGameObject(~name=newName, ());

              _getName() |> expect == newName;
            });
          })
        );

        describe("test rename scene", () =>
          test("ui->scene tree->scene node->name should be updated", () => {
            let newName = "gameObjectNewName";

            SceneTreeInspectorTool.renameGameObject(
              ~name=newName,
              ~gameObject=
                SceneEngineService.getSceneGameObject
                |> StateLogicService.getEngineStateToGetData,
              (),
            );

            BuildComponentTool.buildSceneTree(TestTool.buildEmptyAppState())
            |> ReactTestTool.createSnapshotAndMatch;
          })
        );
      });

      describe("test show Scene inspector", () =>
        test("should show component ui", () => {
          MainEditorSceneTool.addSceneGameObjectComponentTypeToMap();

          MainEditorSceneTreeTool.Select.selectGameObject(
            ~gameObject=
              SceneEngineService.getSceneGameObject(
                StateEngineService.unsafeGetState(),
              ),
            (),
          );

          BuildComponentTool.buildInspectorComponent(
            TestTool.buildEmptyAppState(),
            InspectorTool.buildFakeAllShowComponentConfig(),
          )
          |> ReactTestTool.createSnapshotAndMatch;
        })
      );
    });
  });