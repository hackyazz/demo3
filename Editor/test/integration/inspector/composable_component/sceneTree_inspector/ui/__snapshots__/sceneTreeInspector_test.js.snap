// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SceneTreeInspector prepare currentSelectSource is SceneTree test rename gameObject test rename scene ui->scene tree->scene node->name should be updated 1`] = `
<article
  className="wonder-sceneTree-component"
  id="wonder-sceneTree-component"
>
  <article
    className="wonder-tree"
  >
    <ul
      className="wonder-tree-node"
    >
      <li>
        <div
          className="item-triangle"
          onMouseDown={[Function]}
        >
          <img
            src="./public/img/down.png"
          />
        </div>
        <div
          className="draggable-container  no-drag"
          draggable={false}
          id="sceneTreeNode-0"
          onClick={[Function]}
          onDragEnd={[Function]}
          onDragLeave={[Function]}
          onDragOver={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          style={Object {}}
        >
          <img
            className="treeNode-icon"
            src="./public/img/gameObjectIcon.png"
          />
          gameObjectNewName
        </div>
      </li>
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <div
            className="draggable-container  no-drag"
            draggable={true}
            id="sceneTreeNode-6"
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            <img
              className="treeNode-icon"
              src="./public/img/cameraIcon.png"
            />
            Camera
          </div>
        </li>
      </ul>
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <div
            className="draggable-container select-active no-drag"
            draggable={true}
            id="sceneTreeNode-3"
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            <img
              className="treeNode-icon"
              src="./public/img/gameObjectIcon.png"
            />
            Cube
          </div>
        </li>
      </ul>
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <div
            className="draggable-container  no-drag"
            draggable={true}
            id="sceneTreeNode-4"
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            <img
              className="treeNode-icon"
              src="./public/img/gameObjectIcon.png"
            />
            Cube
          </div>
        </li>
      </ul>
      <ul
        className="wonder-tree-node"
      >
        <li>
          <div
            className="item-triangle"
          />
          <div
            className="draggable-container  no-drag"
            draggable={true}
            id="sceneTreeNode-5"
            onClick={[Function]}
            onDragEnd={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            style={Object {}}
          >
            <img
              className="treeNode-icon"
              src="./public/img/lightIcon.png"
            />
            Direction Light
          </div>
        </li>
      </ul>
    </ul>
  </article>
</article>
`;

exports[`SceneTreeInspector prepare currentSelectSource is SceneTree test show Scene inspector should show component ui 1`] = `
<article
  className="wonder-inspector-component"
>
  <article
    className="wonder-inspector-sceneTree"
  >
    <div
      className="sceneTree-name"
    >
      <article
        className="inspector-item"
      >
        <div
          className="item-header"
          title="gameObject's name"
        >
          <span
            className="component-label"
          >
            Name
          </span>
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="scene"
          />
        </div>
      </article>
    </div>
    <article
      className="componentBox-component"
    >
      <div
        className="header"
      >
        <div
          className="header-triangle"
          onClick={[Function]}
        >
          <span
            className="triangle-bottom"
          />
        </div>
        <div
          className="header-title"
          title="Transform"
        >
          Transform
        </div>
      </div>
      <article
        className="wonder-inspector-transform"
      >
        <article
          className="inspector-item wonder-three-float-input"
        >
          <div
            className="item-header"
            title="position in local coordinate system"
          >
            Position
          </div>
          <div
            className="item-content"
          >
            <article
              className="inspector-item wonder-float-input"
            >
              <div
                className="item-header component-label"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                onMouseUp={[Function]}
                title=""
              >
                X
              </div>
              <div
                className="item-content"
              >
                <input
                  className="input-component float-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="0"
                />
              </div>
            </article>
            <article
              className="inspector-item wonder-float-input"
            >
              <div
                className="item-header component-label"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                onMouseUp={[Function]}
                title=""
              >
                Y
              </div>
              <div
                className="item-content"
              >
                <input
                  className="input-component float-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="0"
                />
              </div>
            </article>
            <article
              className="inspector-item wonder-float-input"
            >
              <div
                className="item-header component-label"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                onMouseUp={[Function]}
                title=""
              >
                Z
              </div>
              <div
                className="item-content"
              >
                <input
                  className="input-component float-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="0"
                />
              </div>
            </article>
          </div>
        </article>
        <article
          className="inspector-item wonder-three-float-input"
        >
          <div
            className="item-header"
            title="euler angle in local coordinate system"
          >
            Rotation
          </div>
          <div
            className="item-content"
          >
            <article
              className="inspector-item wonder-float-input"
            >
              <div
                className="item-header component-label"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                onMouseUp={[Function]}
                title=""
              >
                X
              </div>
              <div
                className="item-content"
              >
                <input
                  className="input-component float-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="0"
                />
              </div>
            </article>
            <article
              className="inspector-item wonder-float-input"
            >
              <div
                className="item-header component-label"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                onMouseUp={[Function]}
                title=""
              >
                Y
              </div>
              <div
                className="item-content"
              >
                <input
                  className="input-component float-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="0"
                />
              </div>
            </article>
            <article
              className="inspector-item wonder-float-input"
            >
              <div
                className="item-header component-label"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                onMouseUp={[Function]}
                title=""
              >
                Z
              </div>
              <div
                className="item-content"
              >
                <input
                  className="input-component float-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="0"
                />
              </div>
            </article>
          </div>
        </article>
        <article
          className="inspector-item wonder-three-float-input"
        >
          <div
            className="item-header"
            title="scale in local coordinate system"
          >
            Scale
          </div>
          <div
            className="item-content"
          >
            <article
              className="inspector-item wonder-float-input"
            >
              <div
                className="item-header component-label"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                onMouseUp={[Function]}
                title=""
              >
                X
              </div>
              <div
                className="item-content"
              >
                <input
                  className="input-component float-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="1"
                />
              </div>
            </article>
            <article
              className="inspector-item wonder-float-input"
            >
              <div
                className="item-header component-label"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                onMouseUp={[Function]}
                title=""
              >
                Y
              </div>
              <div
                className="item-content"
              >
                <input
                  className="input-component float-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="1"
                />
              </div>
            </article>
            <article
              className="inspector-item wonder-float-input"
            >
              <div
                className="item-header component-label"
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                onMouseUp={[Function]}
                title=""
              >
                Z
              </div>
              <div
                className="item-content"
              >
                <input
                  className="input-component float-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value="1"
                />
              </div>
            </article>
          </div>
        </article>
      </article>
    </article>
    <article
      className="wonder-addable-component"
    >
      <div
        className="addable-component-content"
      >
        <div
          className="content-btn"
        >
          <button
            className="addable-btn"
            onClick={[Function]}
          >
            Add Component
          </button>
        </div>
      </div>
    </article>
  </article>
</article>
`;
